sentry-core/src/main/java/io/sentry/core/Scope.java

```diff
@@ -402,9 +413,30 @@ SessionPair startSession() {
       }
       previousSession = session;
 
-      session =
-          new Session(
-              options.getDistinctId(), user, options.getEnvironment(), options.getRelease());
+      String distinctId = null;
+
+      final User finalUser = getUser();
+      //      synchronized (userLock) {
+      if (finalUser != null) {
+        if (finalUser.getId() != null && !finalUser.getId().isEmpty()) {
+          distinctId = finalUser.getId();
+        } else if (finalUser.getEmail() != null && !finalUser.getEmail().isEmpty()) {
```

Comment:
Why do we use the `email` or the `username` as the `distinctId` if the `id` is not set?

Reply:
because all of them are considered to be a unique identifier, in this order.
if there's no id, check for email, no email as well? username.

Reply:
This was the approach taken in the session supported added in the python SDK at least.

Reply:
Thanks for clarifying that 👍. Didn't know